### PR TITLE
Add Batch Renormalisation

### PR DESCRIPTION
**Pull request by abheesht17**
Verified outputs with the Keras 2 implementation: https://colab.research.google.com/gist/abheesht17/f43fd12378d6abb9691d7ed9ab4938a6/batch-renorm-verify-keras-3-implementation.ipynb.

Reference: https://arxiv.org/abs/1702.03275.

### CODE REVIEW

File: keras/src/layers/normalization/batch_normalization_test.py [hunk 1]
```diff
@@ -239,3 +241,198 @@ def test_masked_broadcast_normalization(self):
...
+        actual_output_np = backend.convert_to_numpy(actual_output)
+        self.assertAllClose(actual_output_np, expected_output, atol=1e-5)
+
+        # Verify moving statistics.
+        expected_renorm_mean = (
+            init_renorm_mean * renorm_momentum
+            + batch_mean * (1 - renorm_momentum)
+        )
+        self.assertAllClose(
+            backend.convert_to_numpy(layer.renorm_mean),
```
Comment (hertschuh) [inline keras/src/layers/normalization/batch_normalization_test.py]: nitpick: `assertAllClose` does `convert_to_numpy` for you, so you don't need that.

Same lines 402, 410, 418, 424

File: keras/src/layers/normalization/batch_normalization.py [hunk 2]
```diff
@@ -208,6 +254,41 @@ def build(self, input_shape):
...
         )
 
+        if self.renorm:
+            # In batch renormalization we track the inference moving stddev
+            # instead of the moving variance to more closely align with the
+            # paper. The stddev is initialized as sqrt of the variance
+            # initializer.
+            def moving_stddev_initializer(shape, dtype=None):
+                return ops.sqrt(
+                    self.moving_variance_initializer(shape, dtype=dtype)
```
Comment (hertschuh) [inline keras/src/layers/normalization/batch_normalization.py]: Don't we have the issue that we cannot reuse the initializers, we have to clone them? (same issue lines 270, 280 and 287)

Comment (abheesht17) [reply keras/src/layers/normalization/batch_normalization.py]: Ah, right. We did that for KerasRS. Fixed

File: keras/src/layers/normalization/batch_normalization_test.py (hunk 1 above)
Comment (abheesht17) [reply keras/src/layers/normalization/batch_normalization_test.py]: My bad

File: keras/src/layers/normalization/batch_normalization.py [hunk 3]
```diff
@@ -208,6 +274,43 @@ def build(self, input_shape):
...
+        if self.renorm:
+            # In batch renormalization we track the inference moving stddev
+            # instead of the moving variance to more closely align with the
+            # paper. The stddev is initialized as sqrt of the variance
+            # initializer.
+            def moving_stddev_initializer(shape, dtype=None):
+                return ops.sqrt(
+                    _clone_initializer(
+                        self.moving_variance_initializer(shape, dtype=dtype)
+                    )
```
Comment (hertschuh) [inline keras/src/layers/normalization/batch_normalization.py]: You're trying to clone the resulting tensor, not the initializer (which will be a no-op).

```python
            def moving_stddev_initializer(shape, dtype=None):
                return ops.sqrt(
                    _clone_initializer(self.moving_variance_initializer)(shape, dtype=dtype)
                )
```

Although this would probably be more readable:

```python
            def moving_stddev_initializer(shape, dtype=None):
                cloned = _clone_initializer(self.moving_variance_initializer)
                return ops.sqrt(cloned(shape, dtype=dtype))
```
